expo push notifications guide and example. (#18952)
supabase/supabase: examples/user-management/expo-push-notifications/README.md

1. Start the development server for your project: `npx expo start --dev-client`
1. Scan the QR code shown in the terminal with your physical device.
1. Sign up/in to create a user in Supabase Auth.

## Enhanced security for push notifications

1. Navigate to your [Expo Access Token Settings](https://expo.dev/accounts/_/settings/access-tokens)
1. Create a new token for usage in Supabase Edge Functions
1. Toggle on "Enhanced Security for Push Notifications"
1. Create the local `.env` file: `cp .env.local.example .env.local`
1. In the newly created `.env.local` file, set your `EXPO_ACCESS_TOKEN` value.

## Deploy the Supabase Edge Function

The database webhook handler to send push notifications is located in [supabase/functions/push/index.ts](./supabase/functions/push/index.ts). Deploy the function to your linked project and set the `EXPO_ACCESS_TOKEN` secret.

1. `supabase functions deploy push`
1. `supabase secrets set --env-file .env.local`

## Create the database webhook

Navigate to the [Database Webhooks settings](https://supabase.com/dashboard/project/_/database/hooks) in your Supabase Dashboard.

> 1. Enable and create a new hook.

Reviewer: same feedback as in the docs site file

> 1. Conditions to fire webhook:
>    1.1. Table: `notifications`
>    1.1. Events: `Insert`
> 1. Webhook configuration: Supabase Edge Functions
> 1. Edge Function:
>    1.1. Method: `POST`
>    1.1. Select which edge function to trigger: `push`
>    1.1. Timeout: `1000`
> 1. HTTP Headers:
>    1.1. Content-type: `application/json`
>    1.1. Click "Add new header" > "Add auth header with service key
> 1. Click "Create webhook"

Author: ```suggestion
1. Conditions to fire webhook: Select the `notifications` table and tick the `Insert` event. 
1. Webhook configuration: Supabase Edge Functions.
1. Edge Function: Select the `push` edge function and leave the method as `POST` and timeout as `1000`. 
1. HTTP Headers: Click "Add new header" > "Add auth header with service key and leave Content-type: `application/json`.
1. Click "Create webhook".
```
